src/linear_algebra/dual.lean

```diff
@@ -886,6 +896,36 @@ begin
   exact finrank_range_add_finrank_ker f,
 end
 
+/-- `f.dual_map` is injective if and only if `f` is surjective -/
+@[simp] lemma dual_map_injective_iff (f : V₁ →ₗ[K] V₂) :
+  function.injective f.dual_map ↔ function.surjective f :=
+begin
+  refine ⟨_, λ h, dual_map_injective_of_surjective h⟩,
+  rw [← range_eq_top, ← ker_eq_bot],
+  intro h,
+  apply finite_dimensional.eq_top_of_finrank_eq,
+  rw ← finrank_eq_zero at h,
+  rw [← add_zero (finite_dimensional.finrank K f.range), ← h,
+      ← linear_map.finrank_range_dual_map_eq_finrank_range,
+      linear_map.finrank_range_add_finrank_ker, subspace.dual_finrank_eq],
+end
+
+/-- `f.dual_map` is surjective if and only if `f` is injective -/
+@[simp] lemma dual_map_surjective_iff [finite_dimensional K V₁] (f : V₁ →ₗ[K] V₂) :
+  function.surjective f.dual_map ↔ function.injective f :=
+begin
+  rw [← range_eq_top, ← ker_eq_bot],
+  split,
+  { intro h,
+    rw [← finrank_eq_zero, ← add_right_inj (finite_dimensional.finrank K f.range),
+        add_zero, linear_map.finrank_range_add_finrank_ker,
+        ← linear_map.finrank_range_dual_map_eq_finrank_range, h,
+        finrank_top, subspace.dual_finrank_eq], },
+  { intro h,
+    rw [range_dual_map_eq_dual_annihilator_ker, h],
+    exact submodule.dual_annihilator_bot, },
+end
+
```

Comment:
Can you add one more lemma here; that `f.dual_map` is bijective if `f` is? The proof should be something like `dual_map_injective_iff.and dual_map_surjective_iff`.

Reply:
done